src/lib/web3-instance.ts

```diff
@@ -1,13 +1,13 @@
-import { getConfig } from 'config'
 import Web3 from 'web3'
+import { getWeb3Provider } from 'lib/web3-provider'
 
-let maybeWeb3Provider: InstanceType<typeof Web3.providers.HttpProvider> | null
 let maybeWeb3Instance: Web3 | null
 
+const web3Provider = getWeb3Provider()
```

Comment:
this is why your tests are failing - we want to lazily load this. anything that imports this file will now try and instantiate the provider